app/code/Magento/Customer/Model/ResourceModel/Customer.php

```diff
@@ -129,6 +129,17 @@ protected function _beforeSave(\Magento\Framework\DataObject $customer)
 
         if (!$customer->getEmail()) {
             throw new ValidatorException(__('The customer email is missing. Enter and try again.'));
+        } else {
```

Comment:
No `} else {` is necessary.

Reply:
So shall i merge with previous if or new if clause

Reply:
```php
if (!$customer->getEmail()) {
    throw new ValidatorException(__('The customer email is missing. Enter and try again.'));
}

if (!preg_match("/^([a-z0-9,!\#\$%&'\*\+\/=\?\^_`\{\|\}~-]|[\x{00A0}"
    . "-\x{D7FF}\x{F900}-\x{FDCF}\x{FDF0}-\x{FFEF}])+(\.("
    . "[a-z0-9,!\#\$%&'\*\+\/=\?\^_`\{\|\}~-]|[\x{00A0}-\x{D7FF}"
    . "\x{F900}-\x{FDCF}\x{FDF0}-\x{FFEF}])+)*@([a-z0-9-]|[\x{00A0}"
    . "-\x{D7FF}\x{F900}-\x{FDCF}\x{FDF0}-\x{FFEF}])+(\.([a-z0-9-]|"
    . "[\x{00A0}-\x{D7FF}\x{F900}-\x{FDCF}\x{FDF0}-\x{FFEF}])+)*\."
    . "(([a-z]|[\x{00A0}-\x{D7FF}\x{F900}-\x{FDCF}\x{FDF0}-\x{FFEF}])"
    . "{2,})$/u", $customer->getEmail())) {
    throw new ValidatorException(__('Please enter a valid email address (Ex: johndoe@domain.com).'));
}
```

Could you also tell where have you taken the pattern from?
I remember PR made during Mage Unconference in Cologne last November and want to know if your pattern is compliant with that one.

Reply:
I took it from magento itself, they are validating same thing in jquery. i.e, form validation is there but server validation not there. You can get it here:-
app\code\Magento\Ui\view\base\web\js\lib\validation\rules.js

Don't mind, can you let me why i need to make constant (is that a standard), because i checked magento core code, their they are using variable itself. For reference:-
app\code\Magento\Customer\Controller\Plugin\Account.php
line 51